File: Python/SQLAlchemy_ORM/bll/services/region_service.py
from dal.repositories.region_repository import RegionRepository
from dal.data.database import SessionLocal
from dal.entities.region import Region

class RegionService:
    def __init__(self):
        self.db = SessionLocal()
        self.repo = RegionRepository(self.db)

    def get_by_id(self, id: int):
        entity = self.repo.get_by_id(id)
        if not entity:
            raise ValueError(f"Entity {id} not found")
        return entity

    def get_all(self):
        return self.repo.get_all()

    def create(self, entity: Region) -> Region:
        return self.repo.create(entity)

    def update(self, id: int, entity: Region) -> Region:
        entity = self.repo.update(id, entity)
        if not entity:
            raise ValueError(f"Entity {id} not found for update")
        return entity

    def delete(self, id: int):
        entity = self.repo.delete(id)
        if not entity:
            raise ValueError(f"Entity {id} not found for deletion")
        return entity

File: Node/Sequelize_ORM/src/bll/services/regionService.js
const RegionRepository = require('../../dal/repositories/regionRepository');

class RegionService {
  async getAll() {
    return await RegionRepository.getAll();
  }

  async getById(id) {
    return await RegionRepository.getById(id);
  }

  async create(data) {
    return await RegionRepository.create(data);
  }

  async update(id, data) {
    return await RegionRepository.update(id, data);
  }

  async delete(id) {
    return await RegionRepository.delete(id);
  }
}

module.exports = new RegionService(); 